src/transformers/trainer.py

```diff
@@ -3581,6 +3581,15 @@ def create_model_card(
             library_name = ModelCard.load(model_card_filepath).data.get("library_name")
             is_peft_library = library_name == "peft"
 
+            # Append existing tags in `tags`
+            existing_tags = ModelCard.load(model_card_filepath).data.tags
```

Comment:
This handles a corner case I just found out for Trainers, if you push into a repo that has already some tags, it will overwrite the existing tags with new tags. This block simply cirumvents this by instead of overwriting new tags it will append new tags from existing tags

Reply:
Look OK to me - only comment is that `existing_tags` might be `None`, so you'll need to convert it to a list before adding `tags` if `ModelCard.load(model_card_filepath).data.tags` doesn't have anything set

Reply:
I pushed in 59738c67c273e7a86b13f6648446fff21367151d a simple workaround, I think that way we don't need to convert it to a list as we care about that corner case only if `tags` are explicilty passed + there are some existing tags already